tls/extensions/s2n_extension_type_lists.c

```diff
@@ -63,6 +63,9 @@ static const s2n_extension_type *const client_hello_extensions[] = {
         &s2n_client_signature_algorithms_extension,
         &s2n_client_server_name_extension,
         &s2n_client_alpn_extension,
+        /* We MUST process the NPN extension after the ALPN extension
+         * because NPN is only negotiated if ALPN is not */
+        &s2n_client_npn_extension,
```

Comment:
I'd group them together, just to make it even clearer. Then nobody will accidentally move alpn without reading the comment.

```suggestion

        /* We MUST process the NPN extension after the ALPN extension
         * because NPN is only negotiated if ALPN is not */
        &s2n_client_alpn_extension,
        &s2n_client_npn_extension,
        
```